### ISSUE TITLE
Memory leak In aiokafka consumer

### ISSUE THREAD

**Issue by iamsinghrajat** [aio-libs/aiokafka#628]
Memeory Leak in `getmany()` function of AIOKafkaConsumer. The memory leak occurs when consumer is assigned a topic which has no messages or very few messages ie. at no load. This issue only occurs when there is timeout_ms is set to any value greater than 0. The smaller the value of timeout_ms, faster is memory leak. The issue occurs due to : https://github.com/aio-libs/aiokafka/blob/10bdd29d569f827605789d8c9910c32483b49346/aiokafka/consumer/fetcher.py#L1100
Here we are not removing pending futures so they keep on accumulating in `self._fetch_waiters` and size of `self._fetch_waiters` keep on increasing
**Reproducible example**
```python
import asyncio
from aiokafka import AIOKafkaConsumer
import json
import time
from pympler import tracker

tr = tracker.SummaryTracker()

"""
make sure they are no messages or very few messages in the topic kafka consumer is consuming from. 
"""
async def start_consumer() -> None:
    t1 = time.time()
    consumer = AIOKafkaConsumer(loop=asyncio.get_event_loop(),
                                bootstrap_servers=['kafkaHost:kafkaPort'],
                                group_id='random_id')
    consumer.subscribe(topics=['topic_name'])
    await consumer.start()
    summary_start = tr.create_summary()
    while True:
        data = await consumer.getmany(timeout_ms=1,
                                      max_records=5)
        if data:
            print(data)

        tr.print_diff(summary_start)
        """
        This will show that _asyncio.Future task keep on increasing with each call to getmany() 
        """
        t2 = time.time()
        print((t2 - t1))


asyncio.run(start_consumer())

```

**Comment by ian-sentropy** [comment]
Seeing the same issue, memory is cleared up as soon as a message is sent:
```python
import asyncio
import tracemalloc

from aiokafka import AIOKafkaConsumer, AIOKafkaProducer

tracemalloc.start()

COUNTER = 0
PRIOR_SNAPSHOT = tracemalloc.take_snapshot()


def memory_check():
    global PRIOR_SNAPSHOT
    snapshot = tracemalloc.take_snapshot()
    top_stats = snapshot.compare_to(PRIOR_SNAPSHOT, "lineno")
    for x in top_stats[:5]:
        print(x)
    PRIOR_SNAPSHOT = snapshot


async def publish():
    producer = AIOKafkaProducer()
    async with producer:
        await producer.send_and_wait(topic="my-topic", value="hello world".encode())


async def process_messages():
    counter = 0
    async with AIOKafkaConsumer("my-topic", group_id="gid") as consumer:
        while True:
            counter += 1
            records = await consumer.getmany(max_records=10, timeout_ms=100)

            if counter % 100 == 0:
                print(f"Counter {counter}, memory check:")
                memory_check()

            if counter == 500:
                await publish()

            if not records:
                continue

            print("Received messages, memory check:")
            memory_check()
            break

asyncio.run(process_messages())
```

```
Counter 100, memory check:
<frozen importlib._bootstrap_external>:525: size=80.3 KiB (+80.3 KiB), count=1426 (+1426), average=58 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/site-packages/kafka/protocol/types.py:152: size=39.2 KiB (+39.2 KiB), count=380 (+380), average=106 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/base_events.py:391: size=18.1 KiB (+18.1 KiB), count=137 (+137), average=135 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/stringprep.py:187: size=18.0 KiB (+18.0 KiB), count=1 (+1), average=18.0 KiB
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/site-packages/aiokafka/consumer/fetcher.py:450: size=14.1 KiB (+14.1 KiB), count=101 (+101), average=143 B
Counter 200, memory check:
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/tracemalloc.py:185: size=42.4 KiB (+42.4 KiB), count=678 (+678), average=64 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/tracemalloc.py:113: size=35.4 KiB (+35.4 KiB), count=378 (+378), average=96 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/site-packages/aiokafka/consumer/fetcher.py:450: size=28.2 KiB (+14.1 KiB), count=201 (+100), average=144 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/base_events.py:391: size=31.4 KiB (+13.3 KiB), count=237 (+100), average=136 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/tasks.py:479: size=14.7 KiB (+6560 B), count=189 (+82), average=80 B
Counter 300, memory check:
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/tracemalloc.py:185: size=125 KiB (+82.4 KiB), count=1996 (+1318), average=64 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/site-packages/aiokafka/consumer/fetcher.py:450: size=42.3 KiB (+14.1 KiB), count=301 (+100), average=144 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/base_events.py:391: size=44.7 KiB (+13.3 KiB), count=337 (+100), average=136 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/tasks.py:479: size=21.1 KiB (+6560 B), count=271 (+82), average=80 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/site-packages/kafka/protocol/types.py:152: size=34.8 KiB (-4240 B), count=328 (-53), average=109 B
Counter 400, memory check:
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/site-packages/aiokafka/consumer/fetcher.py:448: size=32.0 KiB (+24.0 KiB), count=1 (+0), average=32.0 KiB
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/site-packages/aiokafka/consumer/fetcher.py:450: size=56.3 KiB (+14.1 KiB), count=401 (+100), average=144 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/base_events.py:391: size=58.0 KiB (+13.3 KiB), count=437 (+100), average=136 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/tracemalloc.py:185: size=136 KiB (+11.3 KiB), count=2177 (+181), average=64 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/base_events.py:666: size=4312 B (-8448 B), count=31 (-66), average=139 B
Counter 500, memory check:
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/site-packages/aiokafka/consumer/fetcher.py:450: size=70.4 KiB (+14.1 KiB), count=501 (+100), average=144 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/base_events.py:391: size=71.2 KiB (+13.3 KiB), count=537 (+100), average=136 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/tracemalloc.py:185: size=143 KiB (+6976 B), count=2286 (+109), average=64 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/tasks.py:479: size=29.4 KiB (+6640 B), count=377 (+83), average=80 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/base_events.py:666: size=7640 B (+3328 B), count=57 (+26), average=134 B
Received messages, memory check:
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/site-packages/aiokafka/consumer/fetcher.py:450: size=0 B (-70.4 KiB), count=0 (-501)
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/base_events.py:391: size=4760 B (-66.6 KiB), count=36 (-501), average=132 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/tracemalloc.py:185: size=130 KiB (-12.9 KiB), count=2079 (-207), average=64 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/tasks.py:479: size=16.5 KiB (-12.9 KiB), count=211 (-166), average=80 B
/Users/ian/.pyenv/versions/3.7.4/lib/python3.7/asyncio/tasks.py:430: size=4240 B (-4000 B), count=53 (-50), average=80 B
```

**Comment by iamsinghrajat** [comment]
https://github.com/aio-libs/aiokafka/pull/629
Raised a pull request regarding this issue. I guess its more of a hack fix.

**Comment by iamsinghrajat** [comment]
Closing this issue. Memory leak fixed with release `0.7.0`
